File: tests/fixtures/languages/python/src/example_domain/example_model/example_record.py
from uuid import UUID


class ExampleRecord:
    def __init__(self, example_id: str, example_active: bool = False) -> None:
        self.example_id = str(UUID(example_id))
        self.example_active = example_active
        self.example_name = None

    def example_action(self) -> None:
        self.example_active = True

File: tests/fixtures/languages/typescript/src/example_domain/example_model/example_record.ts
import crypto from 'node:crypto';

export class ExampleRecord {
    example_name?: string;

    constructor(public readonly example_id: string, public readonly example_active: boolean = false) {}

    example_fingerprint(): string {
        return crypto.createHash('sha1').update(this.example_id).digest('hex');
    }
}
